src/ZkFold/Base/Algebra/Basic/Class.hs

```diff
@@ -64,8 +61,12 @@ class (AdditiveMonoid a, MultiplicativeMonoid a, FromConstant Natural a) => Semi
 
 class (Semiring a, AdditiveGroup a, FromConstant Integer a) => Ring a
 
+data Characteristic = Infinite | Finite Natural
```

Comment:
Where is this datatype used?

Reply:
Nowhere, it was left here after my experiements. I'll remove it.